cypress/integration/analytics.test.js

```diff
@@ -72,6 +72,52 @@ describe('add_to_cart event', () => {
   })
 })
 
+describe('remove_from_cart event', () => {
+  const testRemoveFromCartEvent = (skuId) => {
+    cy.window().then((window) => {
+      const { dataLayer } = window
+
+      const event = dataLayer.find((e) => e.type === 'remove_from_cart')
+
+      expect(event).to.not.be.null
+      expect(event.data).to.have.property('value')
+
+      const item = event.data.items.find((i) => i.item_variant === skuId)
+
+      expect(item).to.not.be.null
+      expect(item).to.have.property('currency')
+      expect(item).to.have.property('item_name')
+    })
+  }
+
+  context('when removing a product from cart', () => {
+    it('adds remove_from_cart event in the data layer', () => {
+      cy.visit(pages.pdp, options)
+      cy.waitForHydration()
+
+      cy.itemsInCart(0)
+
+      // Add item to cart
+      cy.getById('buy-button').click()
+      cy.itemsInCart(1)
+      cy.getById('checkout-button').should('be.enabled')
+      cy.itemsInCart(1)
+
+      // Remove the added item
+      cy.getById('remove-from-cart-button')
+        .click()
+        .then(($btn) => {
+          cy.itemsInCart(0)
+          const skuId = $btn.attr('data-sku')
+
+          testRemoveFromCartEvent(skuId)
+        })
+      cy.getById('cart-item').should('not.exist')
+      cy.getById('checkout-button').should('be.enabled')
```

Comment:
I wonder if we should remove these lines since we already have this logic [here in cart.test.js](https://github.com/vtex-sites/base.store/blob/master/cypress/integration/cart.test.js#L87-L88), and the code here is more related to analytics events.

And also, since we don't have any items in the cart, the checkout button shouldn't be enabled, right?

What do you think?

Reply:
@eduardoformiga Yeah, makes sense. Better leave these to the specific cart test. Also agree with the checkout button test - not sure why it was passing,  though. I also tested with `cy.getById('checkout-button').should('not.be.enabled')` and it also works, all tests green. Anyways, lines removed. Thanks for the heads up :)